python/ray/serve/_private/proxy_state.py

```diff
@@ -30,35 +33,282 @@
 logger = logging.getLogger(SERVE_LOGGER_NAME)
 
 
+class ActorWrapperCallStatus(str, Enum):
+    PENDING = "PENDING"
+    FINISHED_SUCCEED = "FINISHED_SUCCEED"
+    FINISHED_FAILED = "FINISHED_FAILED"
+
+
+class ActorWrapper(ABC):
+    @property
+    @abstractmethod
+    def actor_id(self) -> str:
+        """Return the actor id of the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def start_new_ready_check(self):
+        """Start a new ready check on the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def start_new_health_check(self):
+        """Start a new health check on the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def start_new_drained_check(self):
+        """Start a new drained check on the proxy actor."""
```

Comment:
the drained part is confusing, can you add more details about what exactly the state transition is and how this interacts w/ update_drained?